applications/asset_tracker_v2/src/cloud/cloud_codec/json_common.c

```diff
@@ -643,23 +669,109 @@ int json_common_neighbor_cells_data_add(cJSON *parent,
 			err = op_code_handle(neighbor_cells, JSON_COMMON_ADD_DATA_TO_ARRAY, NULL,
 					     cell, NULL);
 			if (err) {
+				cJSON_Delete(root);
 				cJSON_Delete(neighbor_cells);
 				cJSON_Delete(cell);
 				return err;
 			}
 		}
 
-		err = op_code_handle(parent, JSON_COMMON_ADD_DATA_TO_OBJECT,
+		err = op_code_handle(root, JSON_COMMON_ADD_DATA_TO_OBJECT,
 				     DATA_NEIGHBOR_CELLS_NEIGHBOR_MEAS, neighbor_cells, NULL);
 		if (err) {
+			cJSON_Delete(root);
 			cJSON_Delete(neighbor_cells);
 			return err;
 		}
 	}
+	json_add_obj(parent, DATA_NEIGHBOR_CELLS_ROOT, root);
+
+	data->queued = false;
+	return err;
+}
+
+#if defined(CONFIG_LOCATION_METHOD_WIFI)
+int json_common_wifi_ap_data_add(cJSON *parent,
+				 struct cloud_data_wifi_access_points *data,
+				 enum json_common_op_code op)
+{
+	int err;
+
+	if (!data->queued) {
+		return -ENODATA;
+	}
+
+	cJSON *root = cJSON_CreateObject();
+
+	if (root == NULL) {
+		err = -ENOMEM;
+		return err;
+	}
+
+
+	err = date_time_uptime_to_unix_time_ms(&data->ts);
+	if (err) {
+		LOG_ERR("date_time_uptime_to_unix_time_ms, error: %d", err);
+		cJSON_Delete(root);
+		return err;
+	}
+
+	err = json_add_number(root, DATA_TIMESTAMP, data->ts);
+	if (err) {
+		LOG_ERR("Encoding error: %d returned at %s:%d", err, __FILE__, __LINE__);
+		cJSON_Delete(root);
+		return err;
+	}
+
+	if (data->cnt < NRF_CLOUD_LOCATION_WIFI_AP_CNT_MIN) {
+		cJSON_Delete(root);
+		return -ENODATA;
+	}
+
+	cJSON *ap_array = cJSON_CreateArray();
+
+	if (ap_array == NULL) {
+		cJSON_Delete(root);
+		return err;
+	}
+
+	for (size_t i = 0; i < data->cnt; ++i) {
+		char str_buf[sizeof("112233445566")];
```

Comment:
```suggestion
		char str_buf[AP_STRING_SIZE_MAX];
```
Lets avoid using magic numbers :magic_wand:

Reply:
However, if using a macro nothing is left to the imagination

Reply:
`MAC_ADDRESS_STRING_SIZE_MAX` is probably better